src/widget_window.c

```diff
@@ -98,6 +98,12 @@ GtkWidget *widget_window_create(
 	widget = gtk_window_new(GTK_WINDOW_TOPLEVEL);  
 
 #if HAVE_GTK_LAYER_SHELL
+	if (!getenv("WAYLAND_DISPLAY"))
```

Comment:
Some GTK+ applications run with `GDK_BACKENDS=x11` (mtpaint forces this on itself) ... this is not the right way to check if the Wayland backend is **in use**.

Reply:
btw, WAYLAND_DISPLAY is set under Cage

Reply:
> btw, WAYLAND_DISPLAY is set under Cage
I know, that's why the second test.

Reply:
What's the best way to test the value of `GDK_BACKEND` in the context of the rest of the gtkdialog code? Plain C or glib? I see a mixture here and there.

Reply:
Try https://github.com/puppylinux-woof-CE/gtkdialog/issues/106#issuecomment-991842337